File: hookline/examples/consumers/check.php
<?php
require __DIR__ . '/verify.php';
$s = 'whsec_MfKQ9r8GKYqrTwjUPD8ILPZIo2LaLaSw';
$h = ['webhook-id' => 'msg_p5jXN8AQM9LWM0D4loKWxJek',
      'webhook-timestamp' => '1614265330',
      'webhook-signature' => 'v1,g0hM9SsE+OTPJTGt/tmIKtSyZlE3uFJELVlNIOLJ1OE='];
$b = '{"test": 2432232314}';
$huge = 1000000000;
echo 'valid: ' . var_export(hookline_verify($s, $h, $b, $huge), true) . "\n";
try { hookline_verify($s, $h, '{"test": 2432232315}', $huge); echo "FAIL\n"; }
catch (RuntimeException $e) { echo 'tampered rejected: ' . $e->getMessage() . "\n"; }
try { hookline_verify($s, $h, $b, 1); echo "FAIL\n"; }
catch (RuntimeException $e) { echo 'stale rejected: ' . $e->getMessage() . "\n"; }
$rot = $h; $rot['webhook-signature'] = 'v1,AAAA ' . $h['webhook-signature'];
echo 'rotation: ' . var_export(hookline_verify($s, $rot, $b, $huge), true) . "\n";

File: hookline/examples/consumers/check.rb
require_relative "verify"
S = "whsec_MfKQ9r8GKYqrTwjUPD8ILPZIo2LaLaSw"
H = { "webhook-id" => "msg_p5jXN8AQM9LWM0D4loKWxJek",
      "webhook-timestamp" => "1614265330",
      "webhook-signature" => "v1,g0hM9SsE+OTPJTGt/tmIKtSyZlE3uFJELVlNIOLJ1OE=" }
B = '{"test": 2432232314}'
HUGE = 10**9
puts "valid: #{Hookline.verify(S, H, B, tolerance_seconds: HUGE)}"
begin; Hookline.verify(S, H, '{"test": 2432232315}', tolerance_seconds: HUGE); puts "FAIL"
rescue Hookline::Invalid => e; puts "tampered rejected: #{e.message}"; end
begin; Hookline.verify(S, H, B, tolerance_seconds: 1); puts "FAIL"
rescue Hookline::Invalid => e; puts "stale rejected: #{e.message}"; end
rot = H.merge("webhook-signature" => "v1,AAAA " + H["webhook-signature"])
puts "rotation: #{Hookline.verify(S, rot, B, tolerance_seconds: HUGE)}"
